Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -5158,23 +5158,104 @@ namespace AssetProcessor
 
     void AssetProcessorManager::QueryAbsolutePathDependenciesRecursive(AZ::Uuid sourceUuid, SourceFilesForFingerprintingContainer& finalDependencyList, AzToolsFramework::AssetDatabase::SourceFileDependencyEntry::TypeOfDependency dependencyType)
     {
+        struct PathOrUuid
+        {
+            explicit PathOrUuid(AZ::Uuid uuid)
+                : m_uuid(uuid)
+            {
+            }
+
+            explicit PathOrUuid(AZStd::string path) : m_path(AZStd::move(path))
+            {
+
+            }
+
+            bool IsUuid() const
+            {
+                return !m_uuid.IsNull();
+            }
+
+            AZStd::string GetPath() const
+            {
+                return m_path;
+            }
+
+            AZ::Uuid GetUuid() const
+            {
+                return m_uuid;
+            }
+
+            AZStd::string ToString() const
+            {
+                if (IsUuid())
+                {
+                    return m_uuid.ToString<AZStd::string>(false, false);
+                }
+                else
+                {
+                    return m_path;
+                }
+            }
+
+            bool operator==(const PathOrUuid& rhs) const
+            {
+                return rhs.m_path == m_path && rhs.m_uuid == m_uuid;
+            }
+
+            AZStd::string m_path;
+            AZ::Uuid m_uuid;
+        };
+
+        struct Hasher
+        {
+            size_t operator()(const PathOrUuid& obj) const
+                {
+                size_t seed = 0;
+
+                if (obj.IsUuid())
+                {
+                    AZStd::hash_combine(seed, obj.GetUuid());
+                    return seed;
+                }
+
+                AZStd::hash_combine(seed, obj.GetPath());
+                return seed;
+            }
```

Comment:
indentation is off here

Reply:
The auto formatter was really acting up when I was trying to write this...